src/schema/notifications.ts

```diff
@@ -145,6 +145,26 @@ export class ZimbraNotifications {
 		this.getApolloClient().queryManager.broadcastQueries();
 	};
 
+	// Find the actual folder of the shared folder
+	private findSharedItemId = (item: any) => {
+		let sharedFolderId: any;
+		const cachedData = get(this.cache, 'data.data');
+		const allFolders = Object.keys(cachedData).filter(f =>
+			f.includes('Folder:')
+		);
+		const idSplit = item.id.split(':');
+		for (let folderId in allFolders) {
+			if (
+				cachedData[allFolders[folderId]].ownerZimbraId === idSplit[0] &&
```

Comment:
Store cachedData[allFolders[folderId]] into a variable and use it at multiple places.

Reply:
`allFolders[folderId][ownerZimbraId]` holds same data as `cachedData[allFolders[folderId]].ownerZimbraId `

Reply:
@itsmeprashant As mentioned above, allFolders is an array of string of type `Folder:`.
@avadhutprabhudesai Stored the data in variable and used it everywhere.